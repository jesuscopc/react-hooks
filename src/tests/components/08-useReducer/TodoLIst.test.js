import React from 'react';
const { shallow } = require("enzyme")
const { TodoList } = require("../../../components/08-useReducer/TodoList");
const { demoTodos } = require("../../fixtures/demoTodo");

describe('Sui Test <TodoLIst/>', () => {

  const handleToggle = jest.fn();
  const deleteTask = jest.fn();

  const wrapper =  shallow(
    <TodoList
      todos={ demoTodos }
      handleToggle={ handleToggle }
      deleteTask={ deleteTask }
    />);

  test('should be render correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should be render 2 <TodoLIstItem/>', () => {
    const todoListItem = wrapper.find('TodoLIstItem');
    expect( todoListItem.length ).toBe( demoTodos.length );
    expect( todoListItem.at(0).prop('handleToggle') ).toEqual( expect.any(Function) );
  });
  
})
