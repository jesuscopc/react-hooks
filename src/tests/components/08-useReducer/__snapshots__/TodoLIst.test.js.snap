// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui Test <TodoLIst/> should be render correctly 1`] = `
<ol
  className="list-group list-group-flush"
>
  <TodoLIstItem
    deleteTask={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    task={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoLIstItem
    deleteTask={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    task={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ol>
`;
